relations/src/shielder/withdraw.rs

```diff
@@ -271,45 +128,31 @@ impl<S: State> ConstraintSynthesizer<CircuitField> for WithdrawRelation<S> {
         // Check the merkle proof.
         //------------------------
         check_merkle_proof(
-            self.merkle_root,
-            self.leaf_index,
+            self.merkle_root(),
+            self.leaf_index(),
             old_note.to_bytes()?,
-            self.merkle_path,
-            self.max_path_len,
+            self.merkle_path().cloned().unwrap_or_default(),
```

Comment:
if you really hate cloning, then you can create some empty path before this invocation and in case of `self.merkle_path()` returns `Err`, fall back to `&empty`

Reply:
I know but that would be less esthetic in my eyes.